src/Web/View/Base/AbstractView.php

```
@@ -325,72 +271,227 @@
      *
-     * @param string $extension The extension to remove renderer for.
+     * // Register a renderer with additional definitions
+     * $view = $view->withAddRenderer('twig', TwigRenderer::class, ['cache' => '/tmp/twig-cache']);
+     * ```
      *
-     * @return static View component instance for method chaining.
+     * @phpstan-param class-string<\PHPPress\Renderer\RendererInterface> $renderer
+     * @phpstan-param array<string, mixed> $definition
      */
-    public function removeRenderer(string $extension): static
+    public function withAddRenderer(string $extension, string $renderer, array $definition = []): static
```

PHPStan (2.0.3)

Syntax error, unexpected T_PUBLIC on line 333


(phpstan.parse)